src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { type FC } from 'react';
+
+import searchIcon from './assets/search-icon.svg';
+import { DEFAULT_PLACEHOLDER } from './SearchBar.constants';
+import { useSearchBar } from './SearchBar.hook';
+import { SearchContainer, StyledInput, SearchIconButton } from './SearchBar.styles';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const SearchBar: FC<SearchBarProps> = (props) => {
+  const {
+    placeholder = DEFAULT_PLACEHOLDER,
+    inputOnChange,
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+    disabled,
+    ...otherProps
+  } = props;
+
+  const hookProps = {
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+  };
+
+  const {
+    value: inputValue,
+    handleInputChange,
+    handleKeyDown,
+    handleButtonClick,
+  } = useSearchBar(hookProps);
+
+  const combinedHandleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    // Type assertion para HTMLInputElement
+    const inputEvent = e as React.ChangeEvent<HTMLInputElement>;
+    handleInputChange(inputEvent);
+
+    if (inputOnChange) {
+      inputOnChange(e);
+    }
+  };
+
+  return (
+    <SearchContainer data-testid="search-bar-container">
+      <SearchIconButton
+        onClick={handleButtonClick}
+        aria-label="search"
+        data-testid="search-button"
+        disabled={disabled}
+      >
+        <img src={searchIcon} alt="Search" data-testid="search-icon" />
```

Comment:
The <img> inside the IconButton has alt="Search". Because the button already has an accessible name (aria-label), the image should be decorative to avoid duplicate announcements. Either use an empty alt attribute (alt="") or mark the image aria-hidden.

**Sugerencia:**
```typescript
Make the image decorative: <img src={searchIcon} alt="" aria-hidden="true" /> or use MUI SvgIcon so the icon is accessible via the button label only.
```